woocommerce/woocommerce
# Add marketplace helper subscriptions API

### Changes proposed in this Pull Request:

This is part of the [feature/marketplace-subscriptions](https://github.com/woocommerce/woocommerce/tree/feature/marketplace-subscriptions) project.

This PR adds a new helper API for the My subscriptions page.
For now the API includes only an endpoint to fetch a list of all subscriptions, we will expand it in future PRs.

The subscriptions list PR returns a list of all WCCOM subscriptions the connected store has + a list of installed WCCOM products without a subscription.

For now we want to keep the current PHP based subscriptions page, so this PR isolates the shared subscription list generation code in `WC_Helper::get_subscription_list_data` and uses this method to fetch data for the new API and the legacy UI.

The `MySubscriptions` component is updated to load data from this API and display it. We will need to update this component in future PRs to fully implement the design.
 


### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Get subscriptions data from the API
```curl
curl --location 'http://localhost:8888/wp-json/wc/v3/marketplace/subscriptions'
```
2. The response response code should be _401Unauthorized_
3. Open the [new My subscriptions page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=my-subscriptions)
4. Confirm that your subscriptions are loaded correctly (you must be connected to WCCOM for this to work)
5. Optional: install a products that has a:
 - active subscription
 - without a subscription
 - with an expired subscription
 - a outdated version of a product
6. [Open the old subscriptions list ](http://localhost:8888/wp-admin/admin.php?page=wc-addons&section=helper)and confirm that it still correctly displays data



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add marketplace/subscriptions API.

#### Comment 

</details>

![Screenshot 2023-09-27 at 13 04 20](https://github.com/woocommerce/woocommerce/assets/1199991/e8f66cbf-12a8-4493-b8d2-f856c8b445e1)
